sfcasts/crud-controller.md

```diff
@@ -1,171 +1,90 @@
+
+I'll head back to my terminal and run:
+
+```terminal
+Symfony console make:admin:crud
+```
+
+Once again, let's get one for `Answer` with the defaults stuff... And one for `Topic` with the default stuff. I'll clear my screen... And the last one we need is for `User`.
+
+Beautiful! The only thing it did is three more of these CRUD controller classes, but to make those useful, we need to link to them. So, I'm going to paste three more links here. Let's say `Answers`... And I'll customize each of these links.
+
+Nice! I'll refresh... and look! We now have four different admin sections. All right. I want to look just a little deeper into how this is working behind the scenes. Go to `QuestionCrud` controller and look at its base class. I'll hold "cmd + ctrl" and jump into `AbstractCrudController`. We saw earlier that our dashboard extends an `AbstractDashboardController`. Our CRUD controllers extend an `AbstractCrudController`.
+
+Pretty much everything about how our CRUD controller works is going to be controlled by overriding these configure methods that you see inside of here. We'll learn about all of these as we go along, but on a high level, `configureCrud` is here to help you configure the entire CRUD section. `configureAssets` allows you to control custom CSS and JavaScript in this section, `configureActions` allows you to control the actions you want, which are things like having an index page, having an edit action, or having a delete action. More on that later.
+
+The last important one is `configureFields`, which is going to configure which fields we see on both the index page and the form. But don't worry about those yet. We'll see all of them later.
+
+Then, below this, you can actually see all of the code that executes the pages. So `index` is our index page, `detail` is a detail show page, `edit` is the edit form.
+
+You can see the code that actually runs behind all of this, which is super useful when we're figuring out how to extend things. But... wait a second. If you scroll back up here to the configure methods, a few of these look familiar. Some of them *also* exist in the dashboard based controller class, specifically `configureAssets`, `configureCrud`, `configureActions`, and `configureFilters`. These all live side of `AbstractDashboardController`. There's `configureAssets`, and then further down here, CRUD actions and filters.
+
+So what's going on?
+
+Click "Questions" and look at the URL. It starts with `/admin` and then a bunch of query parameters. Turns out, everything EasyAdmin is handled via a single giant route. It all runs through the `DashboardController` route - the `/route` that's above the index. So when we go to the `QuestionCrudController`, it's actually matching this route here with extra query parameters to say which CRUD controller and which action runs. You can see `crudController` and `crudAction` right there.
```

Comment:
Is `/route` there correct? I can't remember for sure, but I think the "/" was there already, so I left it this time around.